src/stream/src/executor/source/fetch_executor.rs

```diff
@@ -102,13 +109,23 @@ impl<S: StateStore> FsFetchExecutor<S> {
                 )
                 .await?;
             pin_mut!(table_iter);
-
+            let properties = source_desc.source.config.clone();
             while let Some(item) = table_iter.next().await {
                 let row = item?;
                 let split = match row.datum_at(1) {
-                    Some(ScalarRefImpl::Jsonb(jsonb_ref)) => {
-                        SplitImpl::from(FsSplit::restore_from_json(jsonb_ref.to_owned_scalar())?)
-                    }
+                    Some(ScalarRefImpl::Jsonb(jsonb_ref)) => match properties {
+                        risingwave_connector::source::ConnectorProperties::Gcs(_) => {
+                            let split: OpendalFsSplit<OpendalGcs> =
+                                OpendalFsSplit::restore_from_json(jsonb_ref.to_owned_scalar())?;
+                            SplitImpl::from(split)
+                        }
+                        risingwave_connector::source::ConnectorProperties::OpendalS3(_) => {
+                            let split: OpendalFsSplit<OpendalS3> =
+                                OpendalFsSplit::restore_from_json(jsonb_ref.to_owned_scalar())?;
+                            SplitImpl::from(split)
+                        }
```

Comment:
Can we use `OpenFsSplit::<Src>::restore_from_json` to replace the match?

Reply:
Already tried, but `the trait bound `risingwave_connector::source::SplitImpl: From<OpendalFsSplit<Src>>` is not satisfied
the trait `From<OpendalFsSplit<Src>>` is not implemented for `risingwave_connector::source::SplitImpl`

so  I think this match maybe tolerable.

Reply:
I see. This is because we use [macro](https://github.com/risingwavelabs/risingwave/blob/c05c22e905efce7fd0a1e7f0e430a9b95c2cd0cd/src/connector/src/macros.rs#L156) to `impl From xxx for SplitImpl` and there is no generic when defining SplitImpl. This match sounds reasonable to me and not easy to avoid.